File: Database/java/main/com/cdedonder/amezzon/database/data/Pile.java
package com.cdedonder.amezzon.database.data;

import java.io.IOException;

public class Pile extends AbstractDTO {

    private int id;
    private Product product;
    private double sell, buy;

    public Product getProduct() {
        return product;
    }

    public double getSell() {
        return sell;
    }

    public double getBuy() {
        return buy;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setProduct(Product product) {
        this.product = product;
    }

    public void setSell(double sell) {
        this.sell = sell;
    }

    public void setBuy(double buy) {
        this.buy = buy;
    }

    @Override
    public String toJSON() throws IOException {
        return objectMapper.writeValueAsString(this);
    }
}

File: Client/domain/Pile.py
from domain.GenericModel import GenericModel
from domain.Product import Product


class Pile(GenericModel):
    def __init__(self, id=None, product=Product(), amount=None):
        self.id = id
        self.product = product
        self.amount = amount

    def get_id(self):
        return self.id

    def set_id(self, id):
        self.id = id

    def get_product(self):
        return self.product

    def set_product(self, product):
        self.product = product

    def get_amount(self):
        return self.amount

    def set_amount(self, amount):
        self.amount = amount
